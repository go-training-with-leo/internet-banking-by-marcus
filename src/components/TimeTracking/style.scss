@import "assets/scss/mixins.scss";
@import "assets/scss/variables.scss";

.options {
  width: 100%;
  border-radius: 1em;
  background-color: $black-dark;
  padding: 0 0 1em 0;
  position: relative;
  text-align: center;
  display: flex;
  flex-direction: column;

  h1 {
    font-size: 1.1em;
    text-transform: uppercase;
    color: darkgray;
    letter-spacing: 0.2em;
  }

  .slider {
    width: 80%;
    margin: auto;
  }
  .ticks {
    display: flex;
    justify-content: space-between;
    .o_txt {
      flex: 1;
      font-size: 14px;
    }
    .o_on {
      font-weight: 500;
      color: blue;
    }
  }
}
