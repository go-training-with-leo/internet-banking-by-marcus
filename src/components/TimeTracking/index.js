import React from 'react';
import { convertDatetamp } from 'utils/helpers';

import './style.scss';

const TimeTracking = ({ currentTime, min, max }) => {
  const minDateInSeconds = min?.seconds
    ? min.seconds
    : Math.floor(new Date(min) / 1000);
  const maxDateInSeconds = max?.seconds
    ? max.seconds
    : Math.floor(new Date(max) / 1000);

  return (
    <div className='options'>
      <h1>TIME TRACKING</h1>
      <input
        readOnly
        className='slider'
        type='range'
        min={minDateInSeconds}
        max={maxDateInSeconds}
        list='steplist'
        value={currentTime / 1000}
      />

      <div className='range__output' aria-hidden='true' data-tip>
        <div className='range__output-value-track'>
          <div className='range__output-values' data-values />
        </div>
      </div>
      <div className='ticks'>
        <span className='o_txt'>
          {convertDatetamp(minDateInSeconds * 1000)}
        </span>
        <span className='o_txt'>
          {convertDatetamp(((minDateInSeconds + maxDateInSeconds) / 2) * 1000)}
        </span>
        <span className='o_txt'>
          {convertDatetamp(maxDateInSeconds * 1000)}
        </span>
      </div>
    </div>
  );
};

export default TimeTracking;
