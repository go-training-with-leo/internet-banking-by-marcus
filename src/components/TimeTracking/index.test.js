import React from 'react';
import { render } from '@testing-library/react';
import TimeTracking from '.';

describe('Time tracking', () => {
  const currentTime = 10000;
  const min = 5000;
  const max = 15000;
  it('Props valid', () => {
    const tree = render(
      <TimeTracking currentTime={currentTime} min={min} max={max} />
    );

    const timeTracking = tree.container.querySelector('input.slider');

    expect(timeTracking).toHaveAttribute('min', String(Math.floor(min / 1000)));
    expect(timeTracking).toHaveAttribute('max', String(Math.floor(max / 1000)));
    expect(timeTracking).toHaveAttribute('value', String(currentTime / 1000));

    expect(tree).toMatchSnapshot();
  });
});
