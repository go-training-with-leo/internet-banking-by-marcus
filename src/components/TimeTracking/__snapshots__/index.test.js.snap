// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time tracking Props valid 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="options"
      >
        <h1>
          TIME TRACKING
        </h1>
        <input
          class="slider"
          list="steplist"
          max="15"
          min="5"
          readonly=""
          type="range"
          value="10"
        />
        <div
          aria-hidden="true"
          class="range__output"
          data-tip="true"
        >
          <div
            class="range__output-value-track"
          >
            <div
              class="range__output-values"
              data-values="true"
            />
          </div>
        </div>
        <div
          class="ticks"
        >
          <span
            class="o_txt"
          >
            01/01/1970
          </span>
          <span
            class="o_txt"
          >
            01/01/1970
          </span>
          <span
            class="o_txt"
          >
            01/01/1970
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="options"
    >
      <h1>
        TIME TRACKING
      </h1>
      <input
        class="slider"
        list="steplist"
        max="15"
        min="5"
        readonly=""
        type="range"
        value="10"
      />
      <div
        aria-hidden="true"
        class="range__output"
        data-tip="true"
      >
        <div
          class="range__output-value-track"
        >
          <div
            class="range__output-values"
            data-values="true"
          />
        </div>
      </div>
      <div
        class="ticks"
      >
        <span
          class="o_txt"
        >
          01/01/1970
        </span>
        <span
          class="o_txt"
        >
          01/01/1970
        </span>
        <span
          class="o_txt"
        >
          01/01/1970
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
